### PR TITLE
add: answer to question and votes functionality

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
In this PR, I have added the below functionalities:
- Add, Edit, and Delete answers to a particular question
- Add, Edit, and Delete comments on a particular answer of the question
- Upvote, Downvote, and Unvote feature for posts and questions

### Files to check:
- components/card/CardFooter
- components/answer/Answer
- votesServices
- answerServices

### Files to ignore:
- Controllers
- files with minor modifications

### CODE REVIEW

File: src/backend/controllers/AnswerController.js [hunk 1]
```diff
@@ -57,14 +57,14 @@ export const addAnswerHandler = function (schema, request) {
...
+      answerText: answerData,
       username: user.username,
       createdAt: formatDate(),
       updatedAt: formatDate(),
     };
     question.answers.push(answer);
     this.db.questions.update({ _id: questionId }, question);
-    return new Response(201, {}, { answers: question.answers });
+    return new Response(201, {}, { questions: this.db.questions });
   } catch (error) {
```
Comment (Dev-AkshatGupta) [inline src/backend/controllers/AnswerController.js:68]: Its amazing that you made some modifications in the controller for your questions

**Review by Dev-AkshatGupta — APPROVED** [review]
Clean well indented , properly named code . Was unable to find a single suggestion 
You can merge it.

**Review by Vishnu-Aithal — APPROVED** [review]
Good and Readable Code with Abstractions, All the Functionalities are working, Good to merge.
Small Bug - Empty Comments can be made to answers

**Comment by ApurvaSawant11** [comment]
> Good and Readable Code with Abstractions, All the Functionalities are working, Good to merge. Small Bug - Empty Comments can be made to answers

Thanks for reviewing. I will fix this bug in the following PR :).

File: src/features/home/questionSlice.js [hunk 2]
```diff
@@ -134,6 +141,124 @@ export const deleteQueComment = createAsyncThunk(
...
 );
 
+export const addAnswer = createAsyncThunk(
+  "question/addAnswer",
+  async ({ questionId, answerData }, { rejectWithValue }) => {
+    try {
+      const token = localStorage.getItem("Expliqa_token");
+      const { data } = await addAnswerService(questionId, answerData, token);
+      return data;
+    } catch (error) {
```
Comment (VLeads) [inline src/features/home/questionSlice.js:151]: It's great that you have placed all apis at one place !!

**Review by VLeads — APPROVED** [review]
Overall code is good. You can merge

**Pull request merged**